File: OC/显示层动画/WGOCAnimationPrimary/WGOCAnimationPrimary/位置&形状&颜色&渐变动画/WGCombinationViewController.m
//
//  WGCombinationViewController.m
//  WGOCAnimationPrimary
//
//  Created by wanggang on 2019/5/21.
//  Copyright © 2019 bozhong. All rights reserved.
//

#import "WGCombinationViewController.h"

@interface WGCombinationViewController ()

@property (nonatomic, strong) UIButton *btn;

@end

@implementation WGCombinationViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.view.backgroundColor = UIColor.whiteColor;
    self.title = @"综合位置&形状&颜色&渐变动画";
    
    self.btn = [[UIButton alloc] initWithFrame:CGRectMake(-200, 400, SCREEN_WIDTH-20*2, 30)];
    [_btn setTitle:@"点我" forState:UIControlStateNormal];
    [_btn setTitleColor:UIColor.blackColor forState:UIControlStateNormal];
    _btn.backgroundColor = [UIColor cyanColor];
    _btn.alpha = 0;
    [_btn addTarget:self action:@selector(btnClicked) forControlEvents:UIControlEventTouchUpInside];
    [self.view addSubview:_btn];
}

-(void)viewWillAppear:(BOOL)animated{
    [super viewWillAppear:animated];
    
    [UIView animateWithDuration:2 animations:^{
        
        self.btn.alpha = 1;
        self.btn.frame = CGRectMake(20, 400, self.btn.width, self.btn.height);
    }];
}

- (void)btnClicked{
    
    [UIView animateWithDuration:3 animations:^{
        
        self.btn.alpha = 0;
        self.btn.transform = CGAffineTransformMakeRotation(M_PI);
        self.btn.frame = CGRectMake(SCREEN_WIDTH, 0, 0, 0);
        self.btn.alpha = 0;
        self.btn.backgroundColor = UIColor.yellowColor;
    }];
}

@end

File: Swift/显示层初级动画/WGAnimationPrimary/WGAnimationPrimary/位置&形状&颜色&渐变动画/WGCombinationViewController.swift
//
//  WGCombinationViewController.swift
//  WGAnimationPrimary
//
//  Created by wanggang on 2019/4/18.
//  Copyright © 2019 bozhong. All rights reserved.
//

import UIKit

class WGCombinationViewController: UIViewController {

    var loginBtn: UIButton!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        view.backgroundColor = UIColor.white
        
        loginBtn = UIButton(frame: CGRect(x: -200, y: 200, width: WgWith-20*2, height: 30))
        loginBtn.backgroundColor = UIColor.cyan
        loginBtn.setTitle("登录", for: .normal)
        loginBtn.setTitleColor(UIColor.black, for: .normal)
        loginBtn.alpha = 0;
        loginBtn.addTarget(self, action: #selector(loginBtnClicked), for: .touchUpInside)
        view.addSubview(loginBtn)
    }
    
    @objc func loginBtnClicked(){
        
        UIView.animate(withDuration: 2) {
            
            //旋转
            self.loginBtn.transform = CGAffineTransform(rotationAngle: CGFloat(Double.pi))
            //位置,形状
            self.loginBtn.frame = CGRect(x: WgWith, y: 64, width: 0, height: 0)
            //颜色改变
            self.loginBtn.backgroundColor = UIColor.yellow
            //透明度
            self.loginBtn.alpha = 0;
        }
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        
        UIView.animate(withDuration: 2) {
            
            self.loginBtn.frame = CGRect(x: 20, y: self.loginBtn.top, width: self.loginBtn.width, height: self.loginBtn.height)
            self.loginBtn.backgroundColor = UIColor.red
            self.loginBtn.alpha = 1;
        }
    }
}


